qpmodel/optimizer.cs

```diff
@@ -345,11 +405,44 @@ public void ExploreGroup(Memo memo)
         // propagate the property requirement from the top
         public void PropagateProperty(PhysicProperty property)
         {
-            return;
+            // after exploration done
+            Debug.Assert(explored_);
+            Debug.Assert(exprList_.Count >= 2);
+
+            bool solveroptimized = IsSolverOptimizedGroup();
+
+            if ((property != null && propertyCandidates_.ContainsKey(property)) ||
+                (property is null && propertyCandidates_.Count > 0))
+                return;
+            if (property != null)
+                propertyCandidates_.Add(property, new List<int>());
+            if (property is null)
+            {
+                if (propagated_) return;
+                propagated_ = true;
+            }
+
+            for (int i = 0; i < exprList_.Count; i++)
+            {
+                if (exprList_[i].physic_ != null && !solveroptimized)
```

Comment:
is it possible to follow ExplorGroup() style like:
  EnforceGroup()
     foreach (member in exprList_)
         EnforceMember();